px-auto-filter-field.html

```diff
@@ -0,0 +1,139 @@
+<link rel="import" href="../polymer/polymer-element.html"/>
+<link rel="import" href="px-data-grid-theme.html">
+<link rel="import" href="../app-localize-behavior/app-localize-behavior.html"/>
+
+<dom-module id="px-auto-filter-field">
+  <template>
+    <style>
+
+      :host {
+        position: relative;
+        display: block;
+      }
+
+      .field {
+        background-color: var(--px-auto-filter-field-background-color, red);
+      }
+
+      .field {
+        height: 2em;
+        border: none;
+        border-radius: 0;
+        border-width: 0;
+        border-style: solid;
+        border-color: var(--px-auto-filter-field-bottom-border-color, red);
+        border-bottom-width: 1px;
+        padding: 0 .33333rem;
+        color: var(--px-auto-filter-field-text-color, black);
+        background: var(--px-auto-filter-field-background, none);
+        background-size: auto 70%;
+        background-position: left center;
+        padding-left: 20px;
+        outline: 0;
+      }
+
+      .field:hover {
+        color: var(--px-auto-filter-field-text-color--hover, black);
+        border-color: var(--px-auto-filter-field-bottom-border-color--hover, red);
+        background-color: var(--px-auto-filter-field-background-color--hover, red);
+      }
+
+      .field:focus {
+        color: var(--px-auto-filter-field-text-color--focus, black);
+        border-color: var(--px-auto-filter-field-bottom-border-color--focus, red);
+        background-color: var(--px-auto-filter-field-background-color--focus, red);
+      }
+    </style>
+    <input
+      class="field"
+      type="text"
+      name="autofilter"
+      value="{{value::input}}"
+      placeholder="{{placeholder}}"
+    >
+    </input>
+  </template>
+  <script>
+    {
+      /**
+       * Auto filtering field used by px-data-grid
+       *
+       * @memberof Predix
+       * @extends Polymer.Element
+       */
+      class AutoFilterFieldElement extends Polymer.Element {
+        static get is() {
+          return 'px-auto-filter-field';
+        }
+
+        static get properties() {
+          return {
+
+            /**
+             * Placeholder text shown in field
+             */
+            placeholder: {
+              type: String
+            },
+
+            /**
+             * Timeout value in milliseconds
+             */
+            timeout: {
+              type: Number,
+              value: 400
+            },
+
+            value: {
+              type: String
+            }
+          };
+        }
+
+        ready() {
+          super.ready();
+          const input = this.shadowRoot.querySelector('input');
+          input.addEventListener('keypress', event => this._onKeyPress(event));
+          input.addEventListener('change', event => this._onChange(event));
+        }
+
+        _onChange(event) {
+          this._updateValue(true);
+        }
+
+        _onKeyPress(event) {
+          if (event.keyCode === 13) {
```

Comment:
`keyCode` is deprecated, use `event.key` instead. 

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

Also, for people from Japan,  `e.isComposing` is required.

So

```js
if (e.key === 'Enter' && !e.isComposing) {
```